src/lib/containers/GenericCard.tsx

```diff
@@ -419,18 +419,29 @@ export default class GenericCard extends React.Component<
       marginBottom: isHeader ? '0px' : undefined,
       paddingBottom: showFooter || imageFileHandleIdValue ? undefined : '15px',
     }
-
+    const icon:JSX.Element = <>
```

Comment:
Create icon and pass it to the HeaderCard (instead of passing all info required to construct the icon in the HeaderCard)